File: src/main/java/dev/slne/protect/paper/command/commands/protection/ProtectionCommand.java
package dev.slne.protect.paper.command.commands.protection;

import dev.jorel.commandapi.CommandAPICommand;
import dev.jorel.commandapi.arguments.OfflinePlayerArgument;
import dev.slne.protect.paper.dialogs.ProtectionMainDialog;
import org.bukkit.OfflinePlayer;

public class ProtectionCommand extends CommandAPICommand {

    /**
     * Creates a new protection command
     */
    public ProtectionCommand() {
        super("protect");

        withPermission("surf.protect");

        withOptionalArguments(new OfflinePlayerArgument("player").withPermission("surf.protect.others"));

        executesPlayer((player, args) -> {
            final OfflinePlayer target = args.<OfflinePlayer>getOptionalUnchecked("player").orElse(player);
//            final ProtectionMainMenu gui = new ProtectionMainMenu(player, target);
//
//            gui.show(player);
//            ProtectionMainDialogOld.INSTANCE.open(player);
            player.showDialog(ProtectionMainDialog.INSTANCE.mainDialog(player, target));
        });
    }

}

File: src/main/kotlin/dev/slne/surf/protect/paper/command/commands/protection/ProtectionCommand.kt
package dev.slne.surf.protect.paper.command.commands.protection

import dev.jorel.commandapi.kotlindsl.commandTree
import dev.jorel.commandapi.kotlindsl.playerExecutor
import dev.slne.surf.api.paper.inventory.framework.open
import dev.slne.surf.protect.paper.menu.view.protectionMainView
import dev.slne.surf.protect.paper.permission.ProtectPermissionRegistry

fun protectionCommand() = commandTree("protect") {
    withPermission(ProtectPermissionRegistry.PROTECTION_COMMAND)

    playerExecutor { sender, args ->
        protectionMainView.open(sender)
    }
}